kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -346,32 +471,50 @@ func (c *autoHostEndpointController) getAutoHostendpointExpectedIPs(node *libapi
 	return expectedIPs
 }
 
-// generateAutoHostendpointFromNode returns the expected auto hostendpoint to be
-// created from the given node.
-func (c *autoHostEndpointController) generateAutoHostendpointFromNode(node *libapi.Node) *api.HostEndpoint {
-	hepLabels := make(map[string]string, len(node.Labels)+1)
+// generateAutoHostEndpoint returns a HostEndpoint created based on the specific parameters
+func (c *autoHostEndpointController) generateAutoHostEndpoint(node *libapi.Node, templateLabels map[string]string, hepName string, expectedIPs []string, interfaceName string) *api.HostEndpoint {
+	hepLabels := make(map[string]string)
 	for k, v := range node.Labels {
 		hepLabels[k] = v
 	}
+	for k, v := range templateLabels {
+		if _, ok := hepLabels[k]; ok {
+			logrus.Warnf("overwriting duplicate label %s", k)
+		}
+		hepLabels[k] = v
+	}
 	hepLabels[hepCreatedLabelKey] = hepCreatedLabelValue
 
-	return &api.HostEndpoint{
+	hostEndpoint := &api.HostEndpoint{
+		TypeMeta: metav1.TypeMeta{
+			Kind: api.KindHostEndpoint,
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   c.generateAutoHostendpointName(node.Name),
+			Name:   hepName,
 			Labels: hepLabels,
 		},
 		Spec: api.HostEndpointSpec{
-			Node:          node.Name,
-			InterfaceName: "*",
-			ExpectedIPs:   c.getAutoHostendpointExpectedIPs(node),
-			Profiles:      []string{resources.DefaultAllowProfileName},
+			Node:        node.Name,
+			ExpectedIPs: expectedIPs,
+			Profiles:    []string{resources.DefaultAllowProfileName},
 		},
 	}
+
+	if interfaceName != "" {
```

Comment:
I don't think we need this check - `Spec.InterfaceName` is a string field, and so if we don't set it to anything, it will default to `""` anyway.